src/test/java/unit/SpendProfileUser.java

```diff
@@ -0,0 +1,197 @@
+package unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.serialization.userfacing.UFGroup;
+import bio.terra.cli.serialization.userfacing.UFSpendProfileUser;
+import bio.terra.cli.service.SpendProfileManagerService.SpendProfilePolicy;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import harness.TestCommand;
+import harness.TestUsers;
+import harness.baseclasses.ClearContextUnit;
+import harness.utils.SamGroups;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra spend` commands. */
+@Tag("unit")
+public class SpendProfileUser extends ClearContextUnit {
+  // only an owner on the spend profile can disable emails
+  TestUsers spendProfileOwner = TestUsers.chooseTestUserWithOwnerAccess();
+
+  SamGroups trackedGroups = new SamGroups();
+
+  @AfterAll
+  void cleanupOnce() throws IOException, InterruptedException {
+    spendProfileOwner.login();
+
+    // try to disable spend for each group that was created by a method in this class
```

Comment:
why do we need to do this if we're going to try to delete the groups?

Reply:
Because if you don't remove spend access first, the group deletion fails.